src/librepgp/stream-packet.cpp

```diff
@@ -63,6 +63,15 @@ write_uint16(uint8_t *buf, uint16_t val)
     buf[1] = val & 0xff;
 }
 
+void
+write_uint32(uint8_t *buf, uint32_t val)
+{
+    buf[0] = val >> 24;
```

Comment:
You may use `STORE32BE` macros instead of additional function.

Reply:
Please ignore this, missed some inconsistency in our code. Will handle via this issue: https://github.com/rnpgp/rnp/issues/2106